apps/navstar_overlay/src/navstar_overlay_lashup_kv_listener.erl

```diff
@@ -0,0 +1,160 @@
+    {ok, OldState, OldData}.
+
+%%-------------------------------------------------------------------------------------------------------------
+%% State transition
+%% 
+%%                                                                lashup_event
+%%                                                                  +------+
+%%                                                                  |      |
+%%  +--------------+              +----------------+              +-+--------+                +-------------+
+%%  |              |              |                |              |          |                |             |
+%%  |              | lashup_event |                | lashup event |          |   timeout      |             |
+%%  | unconfigured +------------> |    applying    +------------> | batching +--------------> | configuring |
+%%  |              |              |  first_config  |              |          |                |             |
+%%  |              |              |                |              |          | <--------------+             |
+%%  |              |              |                |              |          |  lashup_event  |             |
+%%  +--------------+              +----------------+              +----------+                +-------------+
+%%
+%%--------------------------------------------------------------------------------------------------------------
+
+unconfigured(info, EventContent, Data) ->
+    lager:error("unconfigured"),
+    {next_state, applying_first_config, Data, {next_event, internal, EventContent}}.
+
+applying_first_config(internal, 
+            _EventContent = {lashup_kv_events, Event = #{key := [navstar, overlay, Subnet], ref := Reference}},
+            _Data = #data{ref = Ref}) when Ref == Reference, is_tuple(Subnet), tuple_size(Subnet) == 2 ->
+    lager:error("applying_first_config-internal"),
+    navstar_overlay_configure:start(reply, Event),
+    keep_state_and_data;
+applying_first_config(info, {navstar_overlay_configure, applied_config}, Data) ->
+    lager:error("applying_first_config-info-done"),
+    {next_state, batching, Data};
+applying_first_config(info, _EventContent, _Data) ->
+    lager:error("applying_first_config-info"),
+    {keep_state_and_data, postpone}.
+            
+batching(info, EventContent, _Data) ->
+    lager:error("batching-info"),
+    {keep_state_and_data, {timeout, ?LISTEN_TIMEOUT, {do_configure, EventContent}}};
+batching(timeout, {do_configure, EventContent}, Data) ->
+    lager:error("batching-timeout"),
+    {next_state, configuring, Data, {next_event, internal, EventContent}}.
+
+configuring(internal, {lashup_kv_events, Event = #{key := [navstar, overlay, Subnet], ref := Reference}},
+            Data = #data{ref = Ref}) when Ref == Reference, is_tuple(Subnet), tuple_size(Subnet) == 2 ->
+    lager:error("configuring-internal"),
+    Pid = navstar_overlay_configure:start(noreply, Event),
+    NewData = Data#data{pid = Pid},
```

Comment:
Generally, you make a updated State data named something like Data0, and the new version Data1.